podvm/qcow2/misc-settings.sh

```diff
@@ -83,6 +83,23 @@ TLS_OPTIONS=-cert-file /etc/certificates/tls.crt -cert-key /etc/certificates/tls
 END
 fi
 
+# If DISABLE_CLOUD_CONFIG is not set or not set to true, then add cloud-init.target as a dependency for process-user-data.service
+# so that required files via cloud-config are available before kata-agent starts
+if [ -z "$DISABLE_CLOUD_CONFIG" ] || [ "$DISABLE_CLOUD_CONFIG" != "true" ]
+then
+# Add cloud-init.target as a dependency for process-user-data.service so that
+# required files via cloud-config are available before kata-agent starts    
+    mkdir -p /etc/systemd/system/process-user-data.service.d
+    cat <<END >> /etc/systemd/system/process-user-data.service.d/10-override.conf
+    [Unit]    
+    After=cloud-init.target
+
+    [Service]
+    ExecStart=
+    ExecStart=true
```

Comment:
Yes

Reply:
kata-agent had the dependency on cloud-init, which I move to process-user-data. If cloud-config is not used, process-user-data provisions the file. And if cloud-config is used, process-user-data returns true but adds the cloud-init dep.
Also processing the user-data to update the agent config is useful for both the cases hence it's done before starting kata-agent.

Reply:
yup the agent config should happen in each case. it just itches me a bit that we have to noop a unit file, instead of skipping it. that probably indicates that we are doing something wrong or in a weird way. 

Still, it works and it's just a nitpick. We can look into that in a separate PR.

Reply:
I'll try your suggestion on both Azure and AWS tomorrow. And if this works then we won't have to use noop.

Reply:
With minor modification to remove ordering issue, I was able to use your suggestion and got it successfully working for AWS as well. Thanks